libs/components/src/lib/indoor-maps.component.ts

```diff
@@ -0,0 +1,494 @@
+                                        <app-icon class="text-sm"
+                                            >near_me</app-icon
+                                        >
+                                    </button>
+                                </div>
+                            </li>
+                        </div>
+                    </ul>
+                </div>
+            </div>
+        </div>
+    `,
+    styles: [
+        `
+            :host {
+                height: 100%;
+                width: 100%;
+            }
+
+            mat-spinner {
+                top: 50%;
+                left: 50%;
+                transform: translate(-50%, -50%);
+            }
+        `,
+    ],
+})
+export class IndoorMapsComponent extends AsyncHandler implements OnInit {
+    public map_view_options: any;
+    public map_view_instance: any;
+    public mapsIndoors_instance: any;
+    public googleMaps_instance: any;
+    public mapsIndoors_directions_service_instance: any;
+    public mapsIndoors_directions_renderer_instance: any;
+
+    public live_data_status: string | boolean = 'enabled';
+    public search_result_items: any[];
+    public selected_destination: any = null;
+
+    public loading: boolean;
+    public actions_hashmap: { [id: string]: ViewAction };
+
+    public user_latitude: number | null = null;
+    public user_longitude: number | null = null;
+    public geolocation_error_message: string = '';
+    public route_error_message: string = '';
+    public coordinates: CustomCoordinates | null = null;
+
+    public readonly buildings = this._org.building_list;
+    public readonly building = this._org.active_building;
+    public readonly setBuilding = (b) => (this._org.building = b);
+
+    public readonly levels = combineLatest([
+        this.building,
+        this._state.options,
+    ]).pipe(
+        filter(([_]) => !!_),
+        map(([bld]) => [
+            {
+                id: this._org.building.id,
+                name: 'All Levels',
+            },
+            ...this._org.levelsForBuilding(bld),
+        ])
+    );
+    public levels_list: any[] = [];
+    public buildings_list: Building[] = [];
+
+    /** Custom CSS styles to apply to the map */
+    @Input() public styles: ViewerStyles;
+    /** List of available user actions for the map */
+    @Input() public actions: ViewAction[];
+    /** Custom coordinates to fixate on the map */
+    @Input() public custom_coordinates: CustomCoordinates;
+
+    @ViewChild('searchInput', { static: true }) searchElement: ElementRef;
+    @ViewChild('searchResultItems') searchResults: ElementRef;
+
+    constructor(
+        private _state: ExploreStateService,
+        private _org: OrganisationService,
+        private _snackBar: MatSnackBar
+    ) {
+        super();
+    }
+
+    async ngOnInit() {
+        await this._org.initialised.pipe(first((_) => !!_)).toPromise();
+        this.setBuilding(this._org.building);
+
+        this.subscription(
```

Comment:
This is kind of redundant as you are duplicating the data from the observable.
Use the observable state instead of storing the data with the async pipe in templates and converting the observable to a promise and awaiting the result.
```
<ng-container *ngFor="let lvl of levels | async">
    ...
</ng-container>
```
```
const level_list = await this.levels.pipe(take(1)).toPromise();
```

Reply:
I've now updated this.